backend/src/Equinor.ProjectExecutionPortal.Application/Queries/WorkSurface/GetWorkSurfaceApps/GetWorkSurfaceAppGroupsWithAppsQuery.cs

```diff
@@ -0,0 +1,85 @@
+﻿using AutoMapper;
+using Equinor.ProjectExecutionPortal.Application.Queries.WorkSurfaceAppGroup;
+using Equinor.ProjectExecutionPortal.Application.Services.AppService;
+using Equinor.ProjectExecutionPortal.Domain.Common.Exceptions;
+using Equinor.ProjectExecutionPortal.Domain.Infrastructure;
+using Equinor.ProjectExecutionPortal.Infrastructure;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Equinor.ProjectExecutionPortal.Application.Queries.WorkSurface.GetWorkSurfaceApps;
+
+public class GetWorkSurfaceAppGroupsWithAppsQuery : QueryBase<IList<WorkSurfaceAppGroupDto>>
+{
+    public GetWorkSurfaceAppGroupsWithAppsQuery(Guid workSurfaceId, string? contextExternalId)
+    {
+        WorkSurfaceId = workSurfaceId;
+        ContextExternalId = contextExternalId;
+    }
+
+    public Guid WorkSurfaceId { get; }
+    public string? ContextExternalId { get; }
+
+    public class Handler : IRequestHandler<GetWorkSurfaceAppGroupsWithAppsQuery, IList<WorkSurfaceAppGroupDto>>
```

Comment:
Is the practice to include query and handler in the same file for this project?

Reply:
Yes. I think it makes sense as they share a purpose. Also fewer files to handle.